neuralmonkey/decoders/decoder.py

```diff
@@ -392,14 +399,22 @@ def _attention_decoder(
                     x = linear([inp] + attns, self.embedding_size)
                 else:
                     x = inp
-                # Run the RNN.
 
+                # Run the RNN.
                 cell_output, state = cell(x, state)
-                states.append(state)
-                # Run the attention mechanism.
 
+                # Run the attention mechanism.
                 attns = [a.attention(cell_output) for a in att_objects]
 
+                if conditional_gru:
+                    x_2 = linear(
```

Comment:
Tohle `x_2` se mi líbí ještě míň než `x`, co tam dlí jetě vykopírováno z TF, ale možná bych to nechal na nějaké větší zeslušnění kódu toho dekodéru.